src/IO/S3Common.cpp

```diff
@@ -85,15 +89,107 @@ class AWSLogger final : public Aws::Utils::Logging::LogSystemInterface
     std::unordered_map<String, Poco::Logger *> tag_loggers;
 };
 
+class S3CredentialsProviderChain : public Aws::Auth::AWSCredentialsProviderChain
```

Comment:
Why not  simply use a builder function for `Aws::Auth::AWSCredentialsProviderChain` instead of this inheritance?

Reply:
Because it is protected.